### PR TITLE
Some clang-tidy fixes:

### PR DESCRIPTION
**Pull request by mattgodbolt**
* const parameters in declarations are not recommended;
  "const int x" is the same as "int x" for overload resolution.
  It's fine in definitions though.
* Various minor warnings and unused things.

### CODE REVIEW

**Comment by mattgodbolt** [comment]
More contentious, so separate from other things, and I'll rebase once the others are merged.

File: src/utils/jsonwriter.cpp [hunk 1]
```diff
@@ -391,13 +392,13 @@ void AsmParser::JsonWriter::JsonWriter::write()
...
-                                            const std::unordered_map<std::string_view, std::string_view> aliased_labels,
-                                            const std::unordered_map<std::string_view, std::unordered_set<std::string_view>> used_data_labels)
-: AsmParser::JsonWriter::JsonWriter(out, lines, labels, filter), used_labels(used_labels),
-  used_weak_labels(used_weak_labels), aliased_labels(aliased_labels), used_data_labels(used_data_labels)
+                                            const Filter &filter,
+                                            std::unordered_map<std::string_view, std::unordered_set<std::string_view>>  used_labels,
+                                            std::unordered_map<std::string_view, std::unordered_set<std::string_view>>  used_weak_labels,
+                                            std::unordered_map<std::string_view, std::string_view>  aliased_labels,
+                                            std::unordered_map<std::string_view, std::unordered_set<std::string_view>>  used_data_labels)
+: AsmParser::JsonWriter::JsonWriter(out, lines, labels, filter), used_labels(std::move(used_labels)),
```
Comment (partouf) [inline src/utils/jsonwriter.cpp:400]: I'm not sure these moves are wise. When `-debugdump` is passed, both DebugJsonWriter and JsonWriter are instantiated with the same variables (main.cpp line 101), but they're moved into this and then reused elsewhere...

**Review by partouf — CHANGES_REQUESTED** [review]

**Comment by partouf** [comment]
Rest looks good

File: src/utils/jsonwriter.cpp (hunk 1 above)
Comment (mattgodbolt) [reply src/utils/jsonwriter.cpp:400]: Oh interesting! clang-tidy suggested them, and normally is pretty good about those things! Good spot though!

**Pull request merged**